src/components/modals/SaveCopy.js

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import Modal from "../common/Modal";
+import axios from 'axios';
+import constants, { VERSION } from '../../constants';
+import { getAuthHeader } from "../../utils/Auth";
+
+const SaveCopy = () => {
+    const dispatch = useDispatch();
+    const map = useSelector(state => state.map);
+    const drawings = useSelector(state => state.drawings);
+    const mapLayers = useSelector(state => state.mapLayers);
+    const markers = useSelector(state => state.markers);
+    const { activeDataGroups } = useSelector(state => state.dataGroups);
+    const { isSnapshot } = useSelector(state => state.mapMeta);
+
+    if (!map.name)
+        return <Modal id="saveCopy">
+            <div className="modal-title">Save a copy</div>
+            <div className="modal-content">
+                <div>Please add a title to save the map first</div>
+            </div>
+        </Modal>
+
+    const saveMap = () => {
+        const activeDataGroupsInfo = activeDataGroups.map(group => ({
+            iddata_groups: group.iddata_groups,
+            title: group.title,
+            userGroupId: group.userGroupId
+        }));
+        const saveData = {
+            map: {
+                ...map,
+                gettingLocation: false,
+                name: map.name,
+                currentLocation: null,
+                searchMarker: null,
+            },
+            drawings: drawings,
+            markers: markers,
+            mapLayers: {
+                landDataLayers: mapLayers.landDataLayers,
+                myDataLayers: activeDataGroupsInfo
+            },
+            version: VERSION,
+            name: map.name,
+
+        };
+
+        console.log(saveData)
+
+        const body = {
+            "eid": null,
+            "name": map.name,
+            "data": JSON.stringify(saveData),
+            "isSnapshot": isSnapshot
+        }
+        axios.post(`${constants.ROOT_URL}/api/user/map/save/`, body, getAuthHeader())
+            .then(() => {
+                axios.get(`${constants.ROOT_URL}/api/user/maps/`, getAuthHeader())
+                    .then((response) => {
+                        dispatch({ type: 'POPULATE_MY_MAPS', payload: response.data });
+                        dispatch({
+                            type: 'CLOSE_MODAL',
+                            payload: 'saveCopy'
+                        });
+                        dispatch({ type: 'SAVE_AS_OFF' });
+                        const newMap = response.data[response.data.length - 1];
+
+                        const newMapId = newMap.map.eid;
+                        dispatch({
+                            type: 'SET_MAP_ID',
+                            payload: newMapId
+                        });
+                        const mapData = JSON.parse(newMap.map.data);
+                        mapData.isSnapshot = isSnapshot;
```

Comment:
I don't think this is needed for new maps, because isSnapshot is now always included by the backend

Reply:
Removed